docker-compose.yml

```diff
@@ -10,13 +10,21 @@ services:
       - PORT=8080
     ports:
       - 8080:8080
-  
+    volumes:
+      - ./api:/root/app
+
   ui:
     image: demo-ui
-    build: 
+    build:
       context: ./ui
       dockerfile: Dockerfile
     environment:
       - REACT_APP_API=http://localhost:8080
     ports:
       - 3000:3000
+    volumes:
+      - ./ui:/app
+      - ui-node_modules:/app/node_modules
```

Comment:
Similarly to the API, we're mounting our `./ui` directory as a volume within the UI container. This will (eventually) get us hot-reloading too. The second volume definition on line 27 is for our dependencies (you can ignore that for now kinda, line 26 is more important)